### PR TITLE
drivers: sensors: add veaa_x_3 proportional pressure regulator sensor

### PR DESCRIPTION
**Pull request by jeppenodgaard**
Add driver for Festo VEAA-X-3 series proportional pressure regulator.

The driver assumes that the maximum ADC value matches the maximum output from the device, and that the maximum DAC value matches the maximum input value for the device. External hardware is probably required between the ADC/DAC and the device.

### CODE REVIEW

**Comment by jeppenodgaard** [comment]
I pushed some commits by a mistake to this branch and have removed them again.
@zephyrbot added new reviewers due to this mistake. Sorry about that!

**Review by MaureenHelm — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Needs to be rebased

**Comment by jeppenodgaard** [comment]
> Needs to be rebased

Thanks. Done :)

File: samples/sensor/veaa_x_3/src/main.c
Comment (decsny) [inline samples/sensor/veaa_x_3/src/main.c:13]: perhaps use an alias for this?

File: samples/sensor/veaa_x_3/Kconfig
Comment (decsny) [inline samples/sensor/veaa_x_3/Kconfig:11]: default n is redundant

File: samples/sensor/veaa_x_3/boards/nucleo_h563zi.overlay
Comment (decsny) [inline samples/sensor/veaa_x_3/boards/nucleo_h563zi.overlay:16]: can't the DAC go in the io-channels property as well? and use io-channels-names property?

File: samples/sensor/veaa_x_3/src/main.c
Comment (jeppenodgaard) [reply samples/sensor/veaa_x_3/src/main.c:13]: `..GET_ONE` is more common in `samples/sensor` dir.

File: samples/sensor/veaa_x_3/boards/nucleo_h563zi.overlay
Comment (jeppenodgaard) [reply samples/sensor/veaa_x_3/boards/nucleo_h563zi.overlay:16]: Not sure. Have not thought about that. Do you mean `io-channels = <&adc1 3 &dac1 2 12>;`?

File: samples/sensor/veaa_x_3/Kconfig
Comment (jeppenodgaard) [reply samples/sensor/veaa_x_3/Kconfig:11]: True. Will fix along with potential `io-channels` changes.

**Review by jeppenodgaard — COMMENTED** [review]
Thanks for the review @decsny! I got single question before I do changes.

File: samples/sensor/veaa_x_3/src/main.c
Comment (decsny) [reply samples/sensor/veaa_x_3/src/main.c:13]: Okay, well, I think maybe an alias would be better, but would like @MaureenHelm's opinion since I don't know why the GET_ONE is the standard here if it is

Comment (jeppenodgaard) [reply samples/sensor/veaa_x_3/src/main.c:13]: Alright. I just did a recursive grep and got about 2-3 more matches on GET_ONE.

File: samples/sensor/veaa_x_3/boards/nucleo_h563zi.overlay
Comment (decsny) [reply samples/sensor/veaa_x_3/boards/nucleo_h563zi.overlay:16]: The DAC controller binding does describe io-channel cells and the stm32 dac binding has "output" as a cell, but I guess now I see the DAC header does not actually have a macro to easily get a DAC spec by the DT like for example the adc.h header has.... so I will say what I recommended actually outside the scope of the PR as I didn't realize the DAC subsystem did not have this support fully implemented yet, so nevermind if you don't want to tackle it here

File: samples/sensor/veaa_x_3/src/main.c
Comment (jeppenodgaard) [reply samples/sensor/veaa_x_3/src/main.c:13]: Is this the only thing that remains before you can approve @decsny?

Comment (decsny) [reply samples/sensor/veaa_x_3/src/main.c:13]: I'm not blocking this, I think the sensors and dac maintainers should review now

File: drivers/sensor/veaa_x_3/Kconfig
Comment (kartben) [inline drivers/sensor/veaa_x_3/Kconfig:7]: nit, missing a word:

```suggestion
	bool "VEAA-X-3 pressure regulator driver"
```

File: drivers/sensor/veaa_x_3/veaa_x_3.c
Comment (MaureenHelm) [inline drivers/sensor/veaa_x_3/veaa_x_3.c:98]: The casts shouldn't be necessary

**Review by MaureenHelm — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

File: drivers/sensor/veaa_x_3/veaa_x_3.c
Comment (jeppenodgaard) [reply drivers/sensor/veaa_x_3/veaa_x_3.c]: Thanks! Fixed.

**Comment by jeppenodgaard** [comment]
@MaureenHelm @teburd I rebased to resolve conflicts and removed the unnecessary casts.

**Review by teburd — APPROVED** [review]
Would you be willing to maintain this driver? If so please add yourself (in another PR) as maintainer of this driver alongside @MaureenHelm as co-maintainer.

We are currently in the process of working, ideally, on moving away from fetch+get to read+decode, new drivers are great but I'd love to see them maintained by the authors if possible.

I can't possibly update every driver in tree, and I'd like us to move that way.

**Review by MaureenHelm — APPROVED** [review]

**Comment by jeppenodgaard** [comment]
> Would you be willing to maintain this driver? If so please add yourself (in another PR) as maintainer of this driver alongside @MaureenHelm as co-maintainer.
> 
> We are currently in the process of working, ideally, on moving away from fetch+get to read+decode, new drivers are great but I'd love to see them maintained by the authors if possible.
> 
> I can't possibly update every driver in tree, and I'd like us to move that way.

Makes sense. Created https://github.com/zephyrproject-rtos/zephyr/pull/72962.

Thanks for the re-reviews :)

**Pull request merged**
